crates/bevy_pbr/src/render/pbr.wgsl

```diff
@@ -95,6 +96,12 @@ fn fragment(in: FragmentInput) -> @location(0) vec4<f32> {
         output_color = alpha_discard(material, output_color);
     }
 
+    // fog
+    if (fog.mode != FOG_MODE_OFF && (material.flags & STANDARD_MATERIAL_FLAGS_FOG_ENABLED_BIT) != 0u) {
+        let view_to_world = in.world_position.xyz - view.world_position.xyz;
+        output_color = apply_fog(output_color, view_to_world);
```

Comment:
we should probably pass the position components separately, in case we want to add the height-based density modifications iq talks about in future

Reply:
I agree, though it's also easy to refactor when we actually do that so not a blocker for this PR imo.

Reply:
Is `view_to_world` likely to be reused for some other effect we're interested in? That's one specific scenario in which passing it from the outside could be useful: to avoid hiding the fact that it's used by the `apply_fog()` function and having it be calculated twice.

If that seems unlikely I can calculate it within the function.

Reply:
i think the cost of a single subtraction isn't really worth worrying about either way. i just want the interface so that if somebody wants height-based fog, they can replace the fog shader handle and not have to replace the whole pbr shader top level.

but as superdump says, not blocking, we can always do it later when we have better support for that kind of substitution/plugin functionality.